packages/core/admin/server/src/controllers/admin.ts

```diff
@@ -185,4 +186,46 @@ export default {
 
     return data;
   },
+
+  async getGuidedTourMeta(ctx: Context) {
+    const userService = getService('user');
+    const { id } = ctx.query as GetGuidedTourMeta.Request['query'];
+
+    const isFirstSuperAdminUser = id ? await userService.isFirstSuperAdminUser(id) : false;
+
+    // Check if any content-type schemas have been create on the api:: namespace
+    const contentTypeSchemaNames = Object.keys(strapi.contentTypes).filter((contentTypeUid) =>
+      contentTypeUid.startsWith('api::')
+    );
+    const didCreateContentTypeSchema = contentTypeSchemaNames.length > 0;
+
+    // Check if any content has been created for content-types on the api:: namespace
+    let hasContent = false;
+    for (const contentTypeSchemaName of contentTypeSchemaNames) {
+      const docs = await strapi
+        .documents(contentTypeSchemaName as Internal.UID.ContentType)
+        .count({});
+      if (docs > 0) {
+        hasContent = true;
+        break;
+      }
+    }
```

Comment:
I think we should try to follow our more functional style habites. Maybe use Array.some instead? (You may have picked your current approach to be able to break out of the loop early but I wonder if you tried different things)